doc/api/typescript.md

```diff
@@ -0,0 +1,153 @@
+# Modules: TypeScript
+
+## Enabling
+
+There are two ways to enable runtime TypeScript support in Node.js:
+
+1. For [full support][] of all of TypeScript's syntax and features, including
+   using any version of TypeScript, use a third-party package.
+
+2. For lightweight support, you can use the built-in support for
+   [type stripping][].
+
+## Full TypeScript support
+
+To use TypeScript with full support for all TypeScript features, including
+`tsconfig.json`, you can use a third-party package. These instructions use
+[`tsx`][] as an example but there are many other similar libraries available.
+
+1. Install the package as a development dependency using whatever package
+   manager you're using for your project. For example, with `npm`:
+
+   ```bash
+   npm install --save-dev tsx
+   ```
+
+2. Then you can run your TypeScript code via:
+
+   ```bash
+   npx tsx your-file.ts
+   ```
+
+   Or alternatively, you can run with `node` via:
+
+   ```bash
+   node --import=tsx your-file.ts
+   ```
+
+## Type stripping
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.0 - Early development
+
+The flag [`--experimental-strip-types`][] enables Node.js to run TypeScript
+files that contain only type annotations. Such files contain no TypeScript
+features that require transformation, such as enums or namespaces. Node.js will
+replace inline type annotations with whitespace, and no type checking is
+performed. TypeScript features that depend on settings within `tsconfig.json`,
+such as paths or converting newer JavaScript syntax to older standards, are
+intentionally unsupported. To get fuller TypeScript support, including support
+for enums and namespaces and paths, see [Full TypeScript support][].
+
+The type stripping feature is designed to be lightweight and as fast as
+possible. By intentionally not supporting syntaxes that require JavaScript code
```

Comment:
I think we will support it in the future so I dont think its correct to use performance as motivation right now.
 I'd say that being the feature in early development it intentionally supports the easiest and smallest substet that does not require sourcemaps